### PR TITLE
Read more button added when text overflow on card

### LINKED ISSUE #671: [Feature] Add 'read more' button when text overflows on the card
### Description

Currently, the card looks kinda confusing, if we would a _read more_ button then it would have more clarities. When clicked on the button, it would simply **open the popup** for that specific card.

### Screenshots

image

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [ ] I am willing to work on this issue (optional)
- [ ] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by Isha988**
## Fixes Issue
closes #671 
fixes #671 

## Changes proposed
Read more button is added for the card when the text over flow

## Screenshots
image

### CODE REVIEW

**Review by CBID2 — APPROVED** [review]
Looks good to me! :) LGTM! :)

**Comment by Isha988** [comment]
> Looks good to me! :) LGTM! :)

Just a gentle reminder of adding **gssoc** label if it is also required in pr.

File: components/Cards/Card.tsx [hunk 1]
```diff
@@ -19,7 +24,13 @@ const Card: FC<{ data: IData }> = (props) => {
           </h2>
           <CopyToClipboard url={url} />
         </header>
-        <p className="h-24 w-full overflow-hidden font-sans">{description}</p>
+        <div className="h-[7rem]">
+          <p ref={descriptionRef} className="h-24 w-full overflow-hidden font-sans text-ellipsis line-clamp-4">{description}</p>
+          {
+            (isOverflow) && 
+            <p className="text-sm underline text-violet-600 dark:text-violet-400" >read more...</p>
```
Comment (rupali-codes) [inline components/Cards/Card.tsx:31]:
```suggestion
            <p className="text-sm underline text-violet-600 dark:text-violet-400 text-right" >Read More</p>
```

File: components/Cards/Card.tsx [hunk 2]
```diff
@@ -19,7 +24,13 @@ const Card: FC<{ data: IData }> = (props) => {
           </h2>
           <CopyToClipboard url={url} />
         </header>
-        <p className="h-24 w-full overflow-hidden font-sans">{description}</p>
+        <div className="h-[7rem]">
```
Comment (rupali-codes) [inline components/Cards/Card.tsx:27]:
```suggestion
        <div className="h-[5rem]">
```

Comment (Isha988) [reply components/Cards/Card.tsx:27]: @rupali-codes probably 5rem height can produce some ui misalignment, I am sure that 7rem will work well

**Review by CBID2 — COMMENTED** [review]
Hey @Isha988! It looks great! :) I'm going to wait for  @rupali-codes to respond & then I’ll review your PR again.

File: components/Cards/Card.tsx
Comment (rupali-codes) [inline components/Cards/Card.tsx:1]: could you also add hover effect to it ?

Comment (rupali-codes) [reply components/Cards/Card.tsx:1]: also, can we align it right to paragraph?

Comment (Isha988) [reply components/Cards/Card.tsx:1]: > could you also add hover effect to it ?

Ya sure.

Comment (Isha988) [reply components/Cards/Card.tsx:1]: > also, can we align it right to paragraph?

It is already right-aligned as I committed the align-right change suggested by you.

Comment (Isha988) [reply components/Cards/Card.tsx:1]: @rupali-codes I have added the hover effect on read more button you can check it now

**Review by CBID2 — APPROVED** [review]
Great job @Isha988! 😄 LGTM! 😄

File: components/Cards/Card.tsx
Comment (rupali-codes) [reply components/Cards/Card.tsx:1]: Thanks Isha! just one more small change then you are good to go!

File: components/Cards/Card.tsx [hunk 3]
```diff
@@ -19,7 +24,13 @@ const Card: FC<{ data: IData }> = (props) => {
...
+          <p ref={descriptionRef} className="h-24 w-full overflow-hidden font-sans text-ellipsis line-clamp-4">{description}</p>
+          {
+            (isOverflow) && 
+            <p className="text-sm underline text-violet-600 dark:text-violet-400 text-right hover:text-violet-400 dark:hover:text-violet-300" >read more...</p>
```
Comment (rupali-codes) [inline components/Cards/Card.tsx:31]:
```suggestion
            <p className="text-sm underline text-violet-600 dark:text-violet-400 text-right hover:text-violet-400 dark:hover:text-violet-300" >Read More</p>
```

**Review by rupali-codes — APPROVED** [review]
Perfect, tysm :)

**Pull request merged**